pisek/config/update_config.py

```diff
@@ -162,24 +168,66 @@ def get_subtask_mask(points, subtasks):
     return sub_mask
 
 
-UPDATERS = {"v1": ("v2", update_to_v2)}
-NEWEST_VERSION = "v2"
+def update_to_v3(config: ConfigParser, task_path: str) -> None:
+    contest_type = config.get("task", "contest_type", fallback="kasiopea")
+    if contest_type not in ["kasiopea", "cms"]:
+        raise TaskConfigError(f"Invalid contest_type: '{contest_type}'")
+    config["task"]["use"] = f"@{contest_type}"
+    # TODO: del config["task"]["contest_type"]
+
+    for program_type in ProgramType:
+        config["limits"][f"{program_type}_clock_mul"] = "0"
+        config["limits"][f"{program_type}_clock_min"] = config.get(
+            "limits", f"{program_type}_clock_limit", fallback="360"
+        )
+        maybe_delete_key(config, "limits", f"{program_type}_clock_limit")
+
+    IGNORED_KEYS = [
+        ("task", "tests"),
+        ("tests", "in_mode"),
+        ("tests", "out_mode"),
+        ("tests", "out_format"),
+        ("tests", "online_validity"),
```

Comment:
We'll probably want to support these two keys at some point.

Reply:
Lucky the v3 is experimental and we can change this code :innocent:

Reply:
Given that updating is currently in memory only, I guess there also is no possibility of anyone accidentally deleting data from their config file, so I guess it's fine.